gadget/internal/mkfs.go

```diff
@@ -106,11 +116,19 @@ func mkfsExt4(img, label, contentsRootDir string, deviceSize quantity.Size) erro
 // mkfsVfat creates a VFAT filesystem in given image file, with an optional
 // filesystem label, and populates it with the contents of provided root
 // directory.
-func mkfsVfat(img, label, contentsRootDir string, deviceSize quantity.Size) error {
+func mkfsVfat(img, label, contentsRootDir string, deviceSize, sectorSize quantity.Size) error {
 	// taken from ubuntu-image
+	// TODO: handle sector sizes other than 512, can't create a block size that
+	// is less than the sector size
+
+	// 512B logical sector size by default, unless the specified sector size is
+	// larger than 512, in which case use the sector size
```

Comment:
Looking at https://github.com/dosfstools/dosfstools/blob/e579a7df89bb3a6df08847d45c70c8ebfabca7d2/src/mkfs.fat.c#L1892-L1898 mkfs.vfat just overrides the sector size of the filesystem if needed and prints out a warning. Perhaps we should leave a link to dosfstools source code in the comment.